asteroids_code/BoilerPlate/Asteroids.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Asteroids.hpp"
+//OpenGL include
+#include <GL/glew.h>
+#include <SDL2/SDL_opengl.h>
+
+Asteroids::Asteroids()
+{
+	origin = Vector2(100.0f, 150.0f);
+	asteroid_state = BIG;
+	angle_degree = 25.0f;
+	auto_rotate = 0.0f;
+	asteroids_points.push_back(Vector2(0.0f, 35.0f));
+	asteroids_points.push_back(Vector2(0.0f, 10.0f));
+	asteroids_points.push_back(Vector2(-15.0f, 0.0f));
+	asteroids_points.push_back(Vector2(-25.0f, 0.0f));
+	asteroids_points.push_back(Vector2(-15.0f, -15.0f));
+	asteroids_points.push_back(Vector2(0.0f, -25.0f));
+	asteroids_points.push_back(Vector2(15.0f, -15.0f));
+	asteroids_points.push_back(Vector2(25.0f, -5.0f));
+	asteroids_points.push_back(Vector2(25.0f, 15.0f));
+	asteroids_points.push_back(Vector2(10.0f, 30.0f));
+	asteroids_points_state = asteroids_points;
+}
+
+Asteroids::Asteroids(Vector2& st_position, float st_angle)
+{
+	origin = st_position;
+	angle_degree = st_angle;
+	auto_rotate = 0.0f;
+	asteroid_state = BIG;
+	asteroids_points.push_back(Vector2(0.0f, 35.0f));
+	asteroids_points.push_back(Vector2(0.0f, 10.0f));
+	asteroids_points.push_back(Vector2(-15.0f, 0.0f));
+	asteroids_points.push_back(Vector2(-25.0f, 0.0f));
+	asteroids_points.push_back(Vector2(-15.0f, -15.0f));
+	asteroids_points.push_back(Vector2(0.0f, -25.0f));
+	asteroids_points.push_back(Vector2(15.0f, -15.0f));
+	asteroids_points.push_back(Vector2(25.0f, -5.0f));
+	asteroids_points.push_back(Vector2(25.0f, 15.0f));
+	asteroids_points.push_back(Vector2(10.0f, 30.0f));
+	asteroids_points_state = asteroids_points;
+}
+
+void Asteroids::Render()
+{
+	glLoadIdentity();
+	float real_angle = angle_degree + auto_rotate;
+	glTranslatef(origin.x, origin.y, 0.0f);
+	glRotatef(real_angle, 0.0f, 0.0f, 1.0f);
+	glBegin(GL_LINE_LOOP);
+	for (int x = 0; x < asteroids_points.size(); x++)
+	{
+		glVertex2f(asteroids_points[x].x, asteroids_points[x].y);
+	}
+	glEnd();
+}
+
+void Asteroids::Move()
+{
+	float angle_radian = math_master.DegToRad(angle_degree);
+	origin += Vector2(-asteroid_speed*sin(angle_radian), asteroid_speed*cos(angle_radian));
+}
+
+void Asteroids::Update(const int w_width, const int w_height)
+{
+	Move();
+	auto_rotate += asteroid_rotation;
+	if (auto_rotate > 360)
+	{
+		auto_rotate -= 360;
+	}
+	if (origin.x >= w_width / 2)
```

Comment:
If this is the wrap around, this is UAF. I'd recommend creating a function. Also, if this logic is repeated across your rendered entities, your code might be asking for a refactor.